koku/masu/external/accounts/hierarchy/account_crawler.py

```diff
@@ -0,0 +1,37 @@
+#
+# Copyright 2020 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+"""Account Hierarchy Abstract class."""
+import logging
+
+LOG = logging.getLogger(__name__)
+
+
+class AccountCrawler:
```

Comment:
It looks like you're intending this to be an abstract class. So, you probably want to sub-class this off of `abc.ABC`.

That will ensure the class automatically raises `NotImplemented` exceptions for unimplemented methods.

```suggestion
from abc import ABC

class AccountCrawler(ABC):
```

Reply:
This is a great suggestion @blentz ,  I am working on it now. Just double checking that I did right at the moment. 😂